src/scripts/main.js

```diff
@@ -1 +1,91 @@
 'use strict';
+
+let isClickedLeft = false;
+let isClickedRight = false;
+let timer;
+const notification = document.createElement('div');
+
+notification.setAttribute('data-qa', 'notification');
+
+function resetTimer() {
+  clearTimeout(timer);
+
+  timer = setTimeout(() => {
+    if (!isClickedLeft) {
+      firstPromise();
+    }
+  }, 3000);
+}
+
+document.addEventListener('mousedown', (e) => {
+  if (e.button === 0) {
+    isClickedLeft = true;
+    firstPromise();
+    resetTimer();
+  }
+
+  if (e.button === 2) {
+    isClickedRight = true;
+  }
+
+  if (isClickedLeft !== isClickedRight) {
+    secondPromise();
+  }
+
+  if (isClickedLeft && isClickedRight) {
+    thirdPromise();
+  }
+});
+
+document.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+async function firstPromise() {
+  return new Promise((resolve, reject) => {
+    if (isClickedLeft) {
+      resolve('First promise was resolved');
+    } else {
+      reject(new Error('First promise was rejected'));
+    }
+  })
+    .then((result) => {
```

Comment:
You're adding the 'success' class and removing the 'error' class every time a promise is resolved. This could lead to unnecessary DOM manipulation if the 'success' class is already present or the 'error' class is not present. You might want to check if the classes are present before adding or removing them.